curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/645f7879ebbdb201892e55e1.md

```diff
@@ -26,7 +26,7 @@ assert(code.match(/const\s+userAvatarUrl\s*=/));
 You should assign a `ternary` operator to the `userAvatarUrl`. Your `ternary` should check if `avatar` starts with `/user_avatar/` and return `avatarUrl.concat(avatar)` or `avatar`.
 
 ```js
-assert(code.match(/const\s+userAvatarUrl\s*=\s*avatar\.startsWith\(\s*('|")\/user_avatar\/\1\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/));
+assert(code.match(/const\s+userAvatarUrl\s*=\s*avatar\.(startsWith|test)\(\s*(['"]\s*\/user_avatar\/\2\s*['"])\s*\)\s*\?\s*avatarUrl\.concat\(\s*avatar\s*\)\s*:\s*avatar\s*/));
```

Comment:
I am unable to pass using the test method

<img width="872" alt="Screenshot 2024-01-10 at 7 20 03 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/116fad0e-b740-4552-a6b3-60765a7b1194">


It looks like the test accepts `avatar.test(user_avatar)`
but the correct syntax should be `/^user_avatar/.test(avatar)`

Reply:
@jdwilkin4 should we allow something like this ?

Reply:
> should we allow something like this ?

It is a valid solution.
Plus the description doesn't explicitly mention to use the startWith mention. Plus I don't know where startsWith is taught in the course, besides this project.
So most people with use the test method because that is what was taught to then in earlier projects of the course.

Reply:
@jdwilkin4 i get it now, @jayanth920 dropped a suggestion for you, hope it helps !